### PR TITLE
New Entry: General file format

### PR DESCRIPTION
**Pull request by yangc95**
### Description

Create general file format entry for keyword spreadsheet.

In the future, I think this table can include other file formats including code files like **.js**, **.py**, and so on. I also think it would be good to make separate tables based on the file type (image, text, code, etc.) and have separate headings for each as well. (Good first issue)

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/general/concepts/file-formats/file-formats.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+Title: 'File formats'
+Description: 'File formats'
```
Comment (yangc95) [inline content/general/concepts/file-formats/file-formats.md:3]:
```suggestion
Description: 'File formats determine how information or data, such as text and images, are created, stored, and read.'
```

File: content/general/concepts/file-formats/file-formats.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Tags:
+  - 'Files'
+CatalogContent:
+  - 'paths/code-foundations'
+  - 'paths/computer-science'
+---
+
+File formats determine how information or data, such as text and images, are created, stored, and read. Each file format has a unique extension. For example, a CSV (Comma Separated Values) file has the extension **.csv**.
+
+Certain formats are designed to compress data for fast processing. Others may contain more features, like how PDF (Portable Document Format) files can contain graphs and charts along with table data whereas a CSV file can only contain plain table data.
```
Comment (yangc95) [inline content/general/concepts/file-formats/file-formats.md:16]:
```suggestion
Certain formats are designed to compress data for fast processing. Others may contain more features, like how XLSX (Microsoft Excel) files can contain graphs and charts along with table data whereas a CSV file can only contain plain table data.
```

File: content/general/concepts/file-formats/file-formats.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+Title: 'File formats'
+Description: 'File formats determine how information or data, such as text and images, are created, stored, and read.'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Files'
```
Comment (Dusch4593) [inline content/general/concepts/file-formats/file-formats.md:8]:
```suggestion
  - 'Files'
  - 'Types'
```

File: content/general/concepts/file-formats/file-formats.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+Certain formats are designed to compress data for fast processing. Others may contain more features, like how XLSX (Microsoft Excel) files can contain graphs and charts along with table data whereas a CSV file can only contain plain table data.
+
+Files can also be converted to other formats, however, some features for data display may be lost depending on the type of conversion.
+
+## Types of file formats
+
+File formats exist for different types of files. Some common file types and formats are listed below:
+
+| File Format                                                                             | Extension | Description                                                                                                  |
+| --------------------------------------------------------------------------------------- | --------- | ------------------------------------------------------------------------------------------------------------ |
```
Comment (Dusch4593) [inline content/general/concepts/file-formats/file-formats.md:25]:
```suggestion
| :---------------------------------------------------------------------------------------: | :---------: | ------------------------------------------------------------------------------------------------------------ |
```

File: content/general/concepts/file-formats/file-formats.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+---
+Title: 'File formats'
+Description: 'File formats determine how information or data, such as text and images, are created, stored, and read.'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Files'
+CatalogContent:
+  - 'paths/code-foundations'
```
Comment (Dusch4593) [inline content/general/concepts/file-formats/file-formats.md:10]: Hmmm I'm having a hard time finding a suitable `learn-` link for this one 😂

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@yangc95 Just a few comments/suggestions from my review 😄

File: content/general/concepts/file-formats/file-formats.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+---
+Title: 'File formats'
```
Comment (Dusch4593) [inline content/general/concepts/file-formats/file-formats.md:2]:
```suggestion
Title: 'File Formats'
```

"Title Case"

File: content/general/concepts/file-formats/file-formats.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  - 'paths/computer-science'
+---
+
+File formats determine how information or data, such as text and images, are created, stored, and read. Each file format has a unique extension. For example, a CSV (Comma Separated Values) file has the extension **.csv**.
+
+Certain formats are designed to compress data for fast processing. Others may contain more features, like how XLSX (Microsoft Excel) files can contain graphs and charts along with table data whereas a CSV file can only contain plain table data.
+
+Files can also be converted to other formats, however, some features for data display may be lost depending on the type of conversion.
+
+## Types of file formats
```
Comment (Dusch4593) [inline content/general/concepts/file-formats/file-formats.md:20]: Aren't headers for Docs entries just "Title Case" outright?

Comment (yangc95) [reply content/general/concepts/file-formats/file-formats.md:20]:
```suggestion
## Types of File Formats
```

File: content/general/concepts/file-formats/file-formats.md (hunk 5 above)
Comment (yangc95) [reply content/general/concepts/file-formats/file-formats.md:11]: 😆 what do you think about `learn-the-command-line`?

Comment (Dusch4593) [reply content/general/concepts/file-formats/file-formats.md:11]: Works for me 😃

**Review by KTom101 — COMMENTED** [review]
@yangc95 To me, I didn't see anything that really needs changing. I would just add the learn path to the metadata.

File: content/general/concepts/file-formats/file-formats.md (hunk 5 above)
Comment (yangc95) [reply content/general/concepts/file-formats/file-formats.md:11]:
```suggestion
  - 'learn-the-command-line'
  - 'paths/code-foundations'
```

**Comment by yangc95** [comment]
@Dusch4593 @KTom101 Thanks for all your great 🆓 feedback guys! Please let me know if there's anything else that needs looking over 🙏

**Review by KTom101 — APPROVED** [review]
@yangc95 Just added a comma 😆

**Review by SSwiniarski — APPROVED** [review]
I don't have anything to add, Looks good!

**Comment by ethan-master-coding** [comment]
Do you think we should have a separate table for each type of file (ex. images, programs, videos, etc) to help with expansion?

**Comment by yangc95** [comment]
> Do you think we should have a separate table for each type of file (ex. images, programs, videos, etc) to help with expansion?

Yeah! That would be good.

**Comment by ethan-master-coding** [comment]
@yangc95 I will do it later today with that will also be adding a lot more file formats later today.

**Comment by yangc95** [comment]
Live link: https://www.codecademy.com/resources/docs/general/file-formats

Ty @KTom101 @Dusch4593 @SSwiniarski

**Pull request merged**